### ISSUE TITLE
Default Docker yum repo has been disabled

### ISSUE THREAD

**Issue by jonjhallettuob** [puppetlabs/puppetlabs-kubernetes#378]
## What you expected to happen?

The Kubernetes class should install packages

## What happened?
yum stopped working because the default Puppet-configured docker repo has been disabled.

curl shows the problem:

```
 $ curl https://yum.dockerproject.org/repo/main/centos/7/repodata/repomd.xml
<html>
<head><title>403 Forbidden</title></head>
<body>
<h1>403 Forbidden</h1>
<ul>
<li>Code: AllAccessDisabled</li>
<li>Message: All access to this object has been disabled</li>
<li>RequestId: 1628613E6EA799D0</li>
<li>HostId: XESHuKuCySZPRYv4qRvl43jgzvIgrDO9B7PPE4xZXIsyn+FMXwoQJbmhTpHhdptoBnY+eEiw8Vg=</li>
</ul>
<hr/>
</body>
</html>
```

## How to reproduce it?

Attempt to install a Kubernetes worker with

```
  class { 'kubernetes':
    worker => true,
  }
```

The Hiera configuration is of the form below.

```
kubernetes::kubernetes_version: 1.15.7
kubernetes::kubernetes_package_version: 1.15.7
kubernetes::container_runtime: docker
kubernetes::cni_network_provider: https://raw.githubusercontent.com/coreos/flannel/master/Documentation/kube-flannel.yml
kubernetes::cni_pod_cidr: 10.244.0.0/16
kubernetes::etcd_initial_cluster:[...],[..],[...]
kubernetes::etcd_peers:
- [...]
- [...]
- [...]
kubernetes::etcd_ip: "%{::ipaddress_eth0}"
kubernetes::kube_api_advertise_address: "%{::ipaddress_eth0}"
kubernetes::api_server_count: 3
kubernetes::install_dashboard: true
kubernetes::controller_address: [...]
```

## Anything else we need to know?

The node installed correctly when the following was added to Hiera. These details were taken from the Docker website, https://docs.docker.com/install/linux/docker-ce/centos/.

```
kubernetes::docker_yum_baseurl: https://download.docker.com/linux/centos/7/x86_64/stable
kubernetes::docker_yum_gpgkey: https://download.docker.com/linux/centos/gpg
kubernetes::docker_package_name: docker-ce
kubernetes::docker_version: 19.03.5-3.el7
```

## Versions:
```
# puppet --version
5.5.10

$ kubectl version
[not installed]

$ docker version OR crictl version
[not installed]

# facter os
{
  architecture => "x86_64",
  family => "RedHat",
  hardware => "x86_64",
  name => "CentOS",
  release => {
    full => "7.7.1908",
    major => "7",
    minor => "7"
  },
  selinux => {
    config_mode => "enforcing",
    current_mode => "enforcing",
    enabled => true,
    enforced => true,
    policy_version => "31"
  }
}
$ grep kubernetes Puppetfile
  mod 'puppetlabs-kubernetes', '5.0.0'
```

## Logs:
The relevant output of `puppet agent -t`:

```
Could not update: Execution of '/bin/yum -d 0 -e 0 -y install docker-engine-17.03.1.ce-1.el7.centos' returned 1: One of the configured repositories failed (docker),
and yum doesn't have enough cached data to continue. At this point the only
safe thing yum can do is fail. There are a few ways to work "fix" this:
1. Contact the upstream for the repository and get them to fix the problem.
2. Reconfigure the baseurl/etc. for the repository, to point to a working
upstream. This is most often useful if you are using a newer
distribution release than is supported by the repository (and the
packages for the previous distribution release still work).
3. Run the command with the repository temporarily disabled
yum --disablerepo=docker ...
4. Disable the repository permanently, so yum won't use it by default. Yum
will then just ignore the repository until you permanently enable it
again or use --enablerepo for temporary usage:
yum-config-manager --disable docker
or
subscription-manager repos --disable=docker
5. Configure the failing repository to be skipped, if it is unavailable.
Note that yum will try to contact the repo. when it runs most commands,
so will have to try and fail each time (and thus. yum will be be much
slower). If it is a very temporary problem though, this is often a nice
compromise:
yum-config-manager --save --setopt=docker.skip_if_unavailable=true
failure: repodata/repomd.xml from docker: [Errno 256] No more mirrors to try.
https://yum.dockerproject.org/repo/main/centos/7/repodata/repomd.xml: [Errno 14] HTTPS Error 403 - Forbidden
```

yum prints this error for any action.

**Comment by carabasdaniel** [comment]
Hi @jonjhallettuob,

Thanks for opening this issue, for now please use the 
[docker_yum_baseurl ](https://github.com/puppetlabs/puppetlabs-kubernetes/blob/e2769dfbceec47a43157d2836ac276c76156e3d1/manifests/init.pp#L334) parameter to set the yum base url in your hiera data.

**Comment by binford2k** [comment]
resolved in #436
